modRlp/src/org/aion/rlp/Value.java

```diff
@@ -294,19 +251,12 @@ public boolean isNull() {
         return value == null;
     }
 
-    public boolean isEmpty() {
+    private boolean isEmpty() {
         decode();
-        if (isNull()) {
-            return true;
-        }
-        if (isBytes() && asBytes().length == 0) {
-            return true;
-        }
-        if (isList() && asList().isEmpty()) {
-            return true;
-        }
-        return isString() && asString().equals("");
-
+        return isNull() // null
+                || isBytes() && asBytes().length == 0 // empty byte array
+                || isList() && asList().isEmpty() // empty list
+                || isString() && asString().equals(""); // empty string
```

Comment:
very minor, but `equals` could be replaced here with `String`'s `isEmpty` method